crate_universe/src/context/crate_context.rs

```diff
@@ -396,6 +396,17 @@ impl CrateContext {
                 }
             }
 
+            // Dependency overrides
+            if let Some(dep_overrides) = &crate_extra.dep_overrides {
+                for (crate_name, new_dep) in dep_overrides {
+                    let dep_filter = |dep: &CrateDependency| &dep.id.name == crate_name;
+                    if self.common_attrs.deps.any_matches(dep_filter) {
+                        self.common_attrs.deps.remove_if(dep_filter);
+                        self.common_attrs.extra_deps.insert(new_dep.clone());
```

Comment:
This feels like it should be an atomic replacement, rather than a remove+insert? In particular, it feels like if the old dep was in a select, so should the new dep be, rather than put into an extra_deps list?

I'd suggest instead adding a `replace<T: Clone>(&mut self, old: &T, new: &T)` to `SelectList` which walks all of the relevant `BTreeSet`s doing replacement in each?